File: Codes in C++/Graph/Flood Filling.cpp
class Solution {
public:
    void dfs(vector<vector<int>>& image,vector<vector<int>>& ans, 
    int sr, int sc,int di[],int dj[],int init_color, int newColor)
    {
        int m=image.size();
        int n=image[0].size();
        ans[sr][sc]=newColor;
        for(int i=0;i<4;i++)
        {
         int nrow=sr+di[i];
         int ncol=sc+dj[i];
        if(nrow>=0&&nrow<m&&ncol>=0&&ncol<n&&image[nrow][ncol]==init_color&&ans[nrow][ncol]!=newColor)
         dfs(image,ans,nrow,ncol,di,dj,init_color,newColor);   
        }

    }

    vector<vector<int>> floodFill(vector<vector<int>>& image, 
    int sr, int sc, int color) {
        int m=image.size();
        int n=image[0].size();
        int init_color=image[sr][sc];
        vector<vector<int>> ans=image;
        int di[]={-1,0,+1,0};
        int dj[]={0,+1,0,-1};
        dfs(image,ans,sr,sc,di,dj,init_color,color);
        return ans;
    }
};

File: Codes in Python3/Graph/Flood Filling.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        m=len(image)
        n=len(image[0])
        ans=image
        init_color=image[sr][sc]
        q=[]
        dr=[-1,0,+1,0]
        dc=[0,+1,0,-1]
        q.append((sr,sc))
        ans[sr][sc]=color
        while q:
            r,c=q.pop(0)
            image[r][c]=color
            for i in range(4):
                nrow=r+dr[i]
                ncol=c+dc[i]
                if 0<=nrow<m and 0<=ncol<n and image[nrow][ncol]==init_color and ans[nrow][ncol]!=color:
                    q.append((nrow,ncol))
                    
        return ans

